spec/support/delayed_job.rb

```diff
@@ -87,6 +82,55 @@ def query_jobs(object, method_name)
     end
   end
 
+  # Usage:
+  #    expect(Mailer).to delay_email_delivery_of(:some_email).with(arguments / argument_matchers)
```

Comment:
So you are telling me that all of this could be avoided if we used `deliver_now` instead of `deliver_later`? 😂

Reply:
Well, I am pretty sure 100% of this codebase could have been avoided by employing a call center with operators updating a giant google spreadsheet. :P But we descided to use background jobs, so we need reliable way of testing scheduled jobs.